modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CustomCheckoutStep.java

```diff
@@ -0,0 +1,131 @@
+package com.liferay.client.extension.web.internal.util;
+
+import com.liferay.commerce.constants.CommerceWebKeys;
+import com.liferay.commerce.context.CommerceContext;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.util.CommerceCheckoutStep;
+import com.liferay.portal.catapult.PortalCatapult;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.Http;
+
+import java.util.Locale;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CustomCheckoutStep implements CommerceCheckoutStep {
+
+	public CustomCheckoutStep(
+		boolean active, String actionURL, String label, String name,
+		boolean order, PortalCatapult portalCatapult, String renderURL,
+		boolean sennaDisabled, boolean showControls, boolean visible) {
+
+		_active = active;
+		_actionURL = actionURL;
+		_label = label;
+		_name = name;
+		_order = order;
+		_portalCatapult = portalCatapult;
+		_renderURL = renderURL;
+		_sennaDisabled = sennaDisabled;
+		_showControls = showControls;
+		_visible = visible;
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return LanguageUtil.get(locale, _label);
+	}
+
+	@Override
+	public String getName() {
+		return _name;
+	}
+
+	@Override
+	public boolean isActive(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws Exception {
+
+		return _active;
+	}
+
+	@Override
+	public boolean isOrder() {
+		return _order;
+	}
+
+	@Override
+	public boolean isSennaDisabled() {
+		return _sennaDisabled;
+	}
+
+	@Override
+	public boolean isVisible(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws Exception {
+
+		return _visible;
+	}
+
+	@Override
+	public void processAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		CommerceContext commerceContext =
+			(CommerceContext)actionRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		CommerceOrder commerceOrder = commerceContext.getCommerceOrder();
+
+		_portalCatapult.launch(
+			commerceOrder.getCompanyId(), Http.Method.POST, "",
```

Comment:
I think you actually need an oAuth2ApplicationExternalReferenceCode to communicate with external apps. I think you need to add it in your configuration.